auto_round/export/export_to_gguf/quant_cuda.py

```diff
@@ -0,0 +1,331 @@
+# Copyright (c) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import numpy as np
+
+QK_K = 256
+K_SCALE_SIZE = 12
+GGML_QUANT_SIZES = {
+    "bf16": (1, 2),
+    "q4_0": (32, 2 + 16),
+    "q4_1": (32, 2 + 2 + 16),
+    "q4_k": (256, 2 + 2 + QK_K//2 + 12),
+    "q2_k": (256, 2 + 2 + QK_K//16 + QK_K//4),
+    "q8_0": (32, 2 + 32)
+}
+
+GGML_QUANT_BLOCK = {}
+
+
+
+
+def register_block(name):
+
+    def register(cls):
+        GGML_QUANT_BLOCK[name] = cls
+        return cls
+
+    return register
+
+
+def ggml_quant_cuda(data, ggml_type, scale=None, zp=None, wmin_m=None, d_scale=None, d_wmin_m=None):
+    block_size, type_size = GGML_QUANT_SIZES[ggml_type]
+
+    data = data.to(torch.float32).to("cuda")
+    scale = scale.to("cuda") if scale is not None else scale
+    zp = zp.to("cuda") if zp is not None else zp
```

Comment:
better not  code like this.  Use `device`  arg to set the device, then we could easily extend this to xpu and other devices